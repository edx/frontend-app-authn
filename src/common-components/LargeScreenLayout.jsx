import React from 'react';
import PropTypes from 'prop-types';

import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { getConfig } from '@edx/frontend-platform';

import messages from './messages';

const LargeScreenLayout = (props) => {
  const { intl, children } = props;

  return (
    <div>
      <div className="h-10 d-flex">
        <div className="bc-brand-700 w-10 skew-deg-170" />
        <div className="bc-brand w-35 skew-deg-170" />
        <div className="bc-primary-700 w-10 skew-deg-170" />
        <div className="bc-accent-a w-20 skew-deg-170" />
        <div className="bc-accent-a-light w-25 skew-deg-170" />
      </div>
      <div className="large-screen-background">
        <div>
          <img alt="edx" className="pt-20 pl-3 mw-65 h-65" src={getConfig().LOGO_WHITE_URL} />
        </div>
        <div className="d-flex float-left w-50">
          <div className="pt-100 pl-50">
            <div className="bc-accent-b rotate-deg-11 w-5 h-124" />
          </div>
          <div className="mw-250 pt-100 pl-50">
            <h1 className="text-white mb-0">{intl.formatMessage(messages['start.learning'])}</h1>
            <h1 className="color-accent-a">{intl.formatMessage(messages['with.edx'])}</h1>
          </div>
        </div>
        <div className="float-right w-50">
          { children }
        </div>
      </div>
    </div>
  );
};

LargeScreenLayout.propTypes = {
  intl: intlShape.isRequired,
  children: PropTypes.node.isRequired,
};

export default injectIntl(LargeScreenLayout);
