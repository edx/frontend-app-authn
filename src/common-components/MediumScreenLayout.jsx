import React from 'react';
import PropTypes from 'prop-types';

import MediumScreenHeader from './MediumScreenHeader';

const MediumScreenLayout = (props) => {
  const { children } = props;

  return (
    <>
      <MediumScreenHeader />
      { children }
    </>
  );
};

MediumScreenLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MediumScreenLayout;
