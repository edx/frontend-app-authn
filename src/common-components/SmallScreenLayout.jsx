import React from 'react';
import PropTypes from 'prop-types';

import SmallScreenHeader from './SmallScreenHeader';

const SmallScreenLayout = (props) => {
  const { children } = props;

  return (
    <>
      <SmallScreenHeader />
      <div className="d-flex align-items-center justify-content-center">
        { children }
      </div>
    </>
  );
};

SmallScreenLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SmallScreenLayout;
