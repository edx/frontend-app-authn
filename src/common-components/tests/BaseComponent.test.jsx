import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import SmallScreenLayout from '../SmallScreenLayout';
import MediumScreenLayout from '../MediumScreenLayout';
import LargeScreenLayout from '../LargeScreenLayout';

const prop = (
  <form>
    <div>
      <input type="text" name="username" id="username" />
      <input type="password" name="password" id="password" />
    </div>
    <div>
      <input type="submit" name="submit" value="Submit" />
    </div>
  </form>
);

it('BaseComponent should match small screen layout with snapshot', () => {
  const tree = renderer.create(
    <IntlProvider locale="en">
      <SmallScreenLayout>
        { prop }
      </SmallScreenLayout>
    </IntlProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('BaseComponent should match medium screen layout with snapshot', () => {
  const tree = renderer.create(
    <IntlProvider locale="en">
      <MediumScreenLayout>
        { prop }
      </MediumScreenLayout>
    </IntlProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('BaseComponent should match large screen layout with snapshot', () => {
  const tree = renderer.create(
    <IntlProvider locale="en">
      <LargeScreenLayout>
        { prop }
      </LargeScreenLayout>
    </IntlProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
