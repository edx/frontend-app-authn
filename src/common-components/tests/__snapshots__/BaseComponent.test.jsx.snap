// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseComponent should match large screen layout with snapshot 1`] = `
<div>
  <div
    className="h-10 d-flex"
  >
    <div
      className="bc-brand-700 w-10 skew-deg-170"
    />
    <div
      className="bc-brand w-35 skew-deg-170"
    />
    <div
      className="bc-primary-700 w-10 skew-deg-170"
    />
    <div
      className="bc-accent-a w-20 skew-deg-170"
    />
    <div
      className="bc-accent-a-light w-25 skew-deg-170"
    />
  </div>
  <div
    className="large-screen-background"
  >
    <div>
      <img
        alt="edx"
        className="pt-20 pl-3 mw-65 h-65"
        src="https://edx-cdn.org/v3/default/logo-white.svg"
      />
    </div>
    <div
      className="d-flex float-left w-50"
    >
      <div
        className="pt-100 pl-50"
      >
        <div
          className="bc-accent-b rotate-deg-11 w-5 h-124"
        />
      </div>
      <div
        className="mw-250 pt-100 pl-50"
      >
        <h1
          className="text-white mb-0"
        >
          Start Learning
        </h1>
        <h1
          className="color-accent-a"
        >
          with edX
        </h1>
      </div>
    </div>
    <div
      className="float-right w-50"
    >
      <form>
        <div>
          <input
            id="username"
            name="username"
            type="text"
          />
          <input
            id="password"
            name="password"
            type="password"
          />
        </div>
        <div>
          <input
            name="submit"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`BaseComponent should match medium screen layout with snapshot 1`] = `
<div>
  <div
    className="h-10 bc-brand-700 d-flex"
  >
    <div
      className="bc-brand-700 w-10 skew-deg-170"
    />
    <div
      className="bc-brand w-80 skew-deg-170"
    />
    <div
      className="bc-primary-700 w-10 skew-deg-170"
    />
  </div>
  <div
    className="medium-screen-background"
  >
    <img
      alt="edx"
      className="pt-20 pl-3 mw-65 h-65"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
    />
    <div
      className="d-flex"
    >
      <div
        className="pt-3 pl-5"
      >
        <div
          className="bc-accent-b rotate-deg-11 w-5 h-120 mt-1"
        />
      </div>
      <div
        className="pt-3 pl-5 mw-230"
      >
        <h1
          className="text-white mb-0"
        >
          Start Learning
        </h1>
        <h1
          className="color-accent-a"
        >
          with edX
        </h1>
      </div>
    </div>
  </div>
  <div
    className="100vh w-100"
  >
    <form>
      <div>
        <input
          id="username"
          name="username"
          type="text"
        />
        <input
          id="password"
          name="password"
          type="password"
        />
      </div>
      <div>
        <input
          name="submit"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`BaseComponent should match small screen layout with snapshot 1`] = `
<div>
  <div
    className="w-100 h-10 d-flex bc-brand-700"
  >
    <div
      className="bc-brand-700 w-20 skew-deg-170"
    />
    <div
      className="bc-brand w-80 skew-deg-170"
    />
  </div>
  <div
    className="small-screen-background"
  >
    <img
      alt="edx"
      className="pt-20 pl-3 mw-65 h-65"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
    />
    <div
      className="d-flex"
    >
      <div
        className=" pl-5 pt-20"
      >
        <div
          className="bc-accent-b rotate-deg-11 w-5 h-72 mt-1"
        />
      </div>
      <div
        className="pl-4 pt-20"
      >
        <h1
          className="text-white mb-0"
        >
          Start Learning
        </h1>
        <h1
          className="color-accent-a"
        >
          with edX
        </h1>
      </div>
    </div>
  </div>
  <div>
    <form>
      <div>
        <input
          id="username"
          name="username"
          type="text"
        />
        <input
          id="password"
          name="password"
          type="password"
        />
      </div>
      <div>
        <input
          name="submit"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</div>
`;
